services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/model/CredentialsDto.java

```diff
@@ -0,0 +1,220 @@
+                });
+        setCredentials(credentials);
+
+        setVersion(version);
+        setUpdatedOn(Instant.now());
+    }
+
+    /**
+     * Gets the identifier of the tenant.
+     *
+     * @return The identifier of the tenant.
+     */
+    public String getTenantId() {
+        return tenantId;
+    }
+
+    /**
+     * Sets the identifier of the tenant.
+     *
+     * @param tenantId The tenant's identifier.
+     * @throws NullPointerException if the tenantId is {@code null}.
+     */
+    public void setTenantId(final String tenantId) {
+        this.tenantId = Objects.requireNonNull(tenantId);
+    }
+
+    /**
+     * Gets the identifier of the device.
+     *
+     * @return The identifier of the device.
+     */
+    public String getDeviceId() {
+        return deviceId;
+    }
+
+    /**
+     * Sets the identifier of the device.
+     *
+     * @param deviceId The identifier of the device.
+     * @throws NullPointerException if the deviceId is {@code null}.
+     */
+    public void setDeviceId(final String deviceId) {
+        this.deviceId = Objects.requireNonNull(deviceId);
+    }
+
+    /**
+     * Gets the list of credentials.
+     *
+     * @return The list of credentials.
+     */
+    public List<CommonCredential> getCredentials() {
+        return credentials;
+    }
+
+    /**
+     * Sets the list of credentials.
+     *
+     * @param credentials A list of credentials.
+     */
+    public void setCredentials(final List<CommonCredential> credentials) {
+        this.credentials = credentials;
+    }
+
+    /**
+     * Checks whether the credentials to be updated already have secret ids.
+     *
+     * @return hasSecretIds {@code true} if any credential already has a secret identifier,
+     *                      otherwise {@code false}
+     */
+    @JsonIgnore
+    public boolean hasSecretIds() {
+        return hasSecretIds;
+    }
+
+    /**
+     * Sets whether the credentials to be updated already have secret ids.
+     *
+     * @param hasSecretIds {@code true} if any credential already has a secret identifier,
+     *                         otherwise {@code false}
+     */
+    @JsonIgnore
+    public void setHasSecretIds(final boolean hasSecretIds) {
+        this.hasSecretIds = hasSecretIds;
+    }
+
+    @JsonIgnore
+    private List<? extends CommonSecret> getSecrets(final CommonCredential credential) {
+        if (credential instanceof PasswordCredential) {
+            return ((PasswordCredential) credential).getSecrets();
+        } else if (credential instanceof X509CertificateCredential) {
+            return ((X509CertificateCredential) credential).getSecrets();
+        } else if (credential instanceof PskCredential) {
+            return ((PskCredential) credential).getSecrets();
+        } else {
+            return ((GenericCredential) credential).getSecrets();
+        }
+    }
+
+    @JsonIgnore
+    private <T extends CommonSecret> T generateSecretId(final T secret) {
```

Comment:
this belongs to the `CommonSecret` class, doesn't it?

Reply:
`CommonSecret` class already has a provision to set the id using  `setId(final String id)`. The method `generateSecretId(...)` uses generated uuids as secret ids. IMHO the device registry implementers should be able to choose the nature of the secret ids. Based on this view for me it makes sense to have this method in this class and not in the `CommonSecret`.